prusti-tests/tests/parse/ui/and.stdout

```diff
@@ -23,7 +23,7 @@ use prusti_contracts::*;
 #[prusti::spec_only]
 #[prusti::spec_id = "$(NUM_UUID)"]
 fn prusti_pre_item_test1_$(NUM_UUID)() -> bool {
-    !!((true && true): bool)
+    !!(((true) && (true)): bool)
```

Comment:
Where did we get the extra parentheses from?

Reply:
They came from [here](https://github.com/viperproject/prusti-dev/pull/1271/commits/d4069dd6e86805754cfa48090f88e08bac2090e8), fixing issues if e.g. `lhs` were `foo || bar`. They are technically avoidable, but I think that would require quite some restructuring to apply initial parentheses to any "untrusted" tokens, then respect precedence levels in the further desugaring. Probably not worth it?

Reply:
I see the issue now. Yes, we could somehow detect when the parentheses are not needed, but that's indeed more effort than it's worth.